WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -248,6 +261,27 @@ final class DashboardViewModel {
         viewModel?.$showWebViewSheet.assign(to: &self.$showWebViewSheet)
         announcementViewModel = viewModel
     }
+
+    /// Sets up observer to decide store onboarding task lists visibility
+    ///
+    private func setupObserverForShowOnboarding() {
+        guard featureFlagService.isFeatureFlagEnabled(.dashboardOnboarding) else {
+            return
+        }
+
+        let noOnboardingTasksAvailablePublisher: AnyPublisher<Bool, Never>
+        if let storeOnboardingViewModel {
+            noOnboardingTasksAvailablePublisher = storeOnboardingViewModel.$noTasksAvailableForDisplay.eraseToAnyPublisher()
+        } else {
+            noOnboardingTasksAvailablePublisher = Just(false).eraseToAnyPublisher()
+        }
+
+        self.showOnboardingCancellable = Publishers.CombineLatest(noOnboardingTasksAvailablePublisher,
+                                                                  userDefaults.publisher(for: \.completedAllStoreOnboardingTasks))
+        .sink { [weak self] noTasksAvailableForDisplay, completedAllStoreOnboardingTasks in
+            self?.showOnboarding = !(noTasksAvailableForDisplay || completedAllStoreOnboardingTasks)
+        }
```

Comment:
nit: I wonder if it makes sense for this combineLatest logic to be in `StoreOnboardingViewModel` since it already takes in `UserDefaults` and it's more of the `StoreOnboardingViewModel`'s responsibility than `DashboardViewModel`'s?

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/9185/commits/8fee921685ba23f6b0808b6f669b30bdbd8d44ed

We still have to do the `userDefaults` check in `DashboardViewModel` to avoid creating `StoreOnboardingViewModel` when not needed.